solutions/cf-1555B.mdx

```diff
@@ -13,29 +13,49 @@ Oftentimes, problems will try to trick you. For example, since the judge is aski
-
-Note: when placing the second rectangle on the left and right sides, make sure $w_{1} + w_{2} \leq W$, otherwise, the second rectangle won't fit.
-
-Also, when placing the second rectangle on the top and bottom sides, make sure $h_{1} + h_{2} \leq H$, otherwise, the second rectangle won't fit either of them.
-
-Calculate these placements, and chose the one that requires the least distance.
+Let's call:
+* $x_{1}$ and $x_{2}$ the leftmost and rightmost points of table 1 respectively
+* $y_{1}$ and $y_{2}$ the bottommost and topmost points of table 1 respectively
+* $w_{1}$ and $h_{1}$ the width and height of table 1 (equal to the difference in $x_{1}$ and $x_
+  {2}$ and $y_{1}$ and $y_{2}$ respectively)
+* $w_{2}$ and $h_{2}$ the width and height of table 2 respectively
+* $W$ and $H$ the width and height of the room respectively
+
+The key to this problem is realising that table 2 will only ever be in 4 different positions
+relative to table 1: to its left, right, top or bottom (see diagram below). In addition, in all
+cases we can assume that table 2 will be touching its respective wall in order to (attempt) to
+maximise its distance from table 1. E.g., if placing table 2 to the left of table 1, we will
+assume its leftmost edge will be touching the left wall of the room. Determining the solution is
+then simply a case of simulating all of these cases and finding which involves moving table 1 the smallest distance.
+
+![](cf-1555B\CF1555B-1.png)
+Possible placements of table 2 relative to table 1.
+
+To determine the minimum distance that table 1 must be moved, we have to calculate the overlap
+between the two tables. For example:
+
+![](cf-1555B\CF1555B-2.png)
+
+In this case, table 2 is placed to the left of table 1. We can therefore calculate its overlap
+with table 1 as follows: $w$ - $x_{1}$. Here the overlap is 1, meaning we
+must move table 1 one space from its starting position to fit table 2 in the room. We must then
+compare this answer to the answers we get from placing table 2 to the right, top or bottom of
+table 1. The smallest of all of these is our final answer.
+
+To summarise, we calculate the overlap of the two tables as follows:
+* Table 2 placed to the left of table 1: Overlap = $w$ - $x_{1}$
+* Table 2 placed to the right of table 1: Overlap = $x_{2}$ - ($W$ - $w$)
+* Table 2 placed to the top of table 1: Overlap = $y_{2}$ - ($H$ - $h$)
+* Table 2 placed to the bottom of table 1: Overlap = $h$ - $y_{1}$
+
+It is possible that the two tables don't overlap at all. In this case the above calculations
+would return a negative number so we should instead return 0 to indicate that table 1 doesn't
+have to move at all.
+
+Note: when placing table 2 on the left and right sides, make sure $w_{1} + w_{2} \leq W$,
+otherwise table 2 won't fit in the room.
+Similarly, when placing table 2 on the top and bottom sides, make sure $h_{1} + h_{2} \leq H$,
+otherwise table 2 won't fit in the room.
```

Comment:
```suggestion
<Info>
When placing table 2 on the left and right sides, make sure $w_{1} + w_{2} \leq W$,
otherwise table 2 won't fit in the room.
Similarly, when placing table 2 on the top and bottom sides, make sure $h_{1} + h_{2} \leq H$,
otherwise table 2 won't fit in the room.
</Info>
```